### PR TITLE
Join team

### PR DESCRIPTION
**Pull request by lichab**
- [x] Create Join Team Page to handle login verification and join team request
- [x] Wrapped Google Login in Login button that handles login settings and service call
- [x] Create renderWithMemoryRouter test util
- [x] Create createAxiosError test util
- [x] Introduced PageRoutes enum

### CODE REVIEW

**Review by bhoang93 — COMMENTED** [review]
looks good but it has conflicts with main now

**Comment by lichab** [comment]
Fixed conflicts! @bhoang93

File: src/components/LoginButton/LoginButton.test.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    beforeEach(() => {
+        render(
+            <BrowserRouter>
+                <LoginButton onLogin={handleLoginRedirection} />
+            </BrowserRouter>
+        );
+    });
+
+    test("renders log in button", () => {
+        const button = screen.getByText(LOGIN_BUTTON_TEXT);
```
Comment (simonbesteiro) [inline src/components/LoginButton/LoginButton.test.tsx:43]: This line can be extracted into before each

File: src/pages/HomePage/HomePage.tsx [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
 
-  const googleLoginHandler = async (
-    res: GoogleLoginResponse | GoogleLoginResponseOffline
-  ) => {
-    const route = await loginService(res);
-    navigate(route);
+  const redirectTo = (isSuccessful: boolean) => {
+    if (isSuccessful) {
+      navigate(PageRoutes.TEAMS);
+      return;
```
Comment (simonbesteiro) [inline src/pages/HomePage/HomePage.tsx:15]: is better have return than a else?

**Review by bhoang93 — DISMISSED** [review]
I don't believe the new component we have (ActivityModal) is using the page routes to go to the error page. We could try to enforce the use of the pageRoutes if we wrap the navigate function but we could do that in another branch.

File: src/pages/HomePage/HomePage.tsx (hunk 2 above)
Comment (lichab) [reply src/pages/HomePage/HomePage.tsx:15]: For me it is,  I can change it if you don't agree

Comment (simonbesteiro) [reply src/pages/HomePage/HomePage.tsx:15]: I don't mind keeping it like this.

Comment (simonbesteiro) [reply src/pages/HomePage/HomePage.tsx:15]: Maybe we can start to apply this all around the code so we follow the same rules

Comment (lichab) [reply src/pages/HomePage/HomePage.tsx:15]: 👍

File: src/components/LoginButton/LoginButton.test.tsx (hunk 1 above)
Comment (lichab) [reply src/components/LoginButton/LoginButton.test.tsx]: changed!

**Review by simonbesteiro — DISMISSED** [review]

**Review by sofiacarballo-codurance — APPROVED** [review]

**Pull request merged**